### PR TITLE
Feature/Error and Success Pop Ups

### PR DESCRIPTION
**Pull request by brendobrendo**
---

- [x] The changes don't come from your master branch.

---

#### What kind of change does this PR introduce? (bug fix, feature, docs update, ...)
- [x] New feature (non-breaking change which adds functionality)

Added the ErrorPopUp and SuccessPopUp components. Both disappear when the "Okay" button is clicked. A route will need to be added for the "Undo" Button on the SuccessPopUp.

Screen Shot 2022-11-26 at 5 30 58 PM

Screen Shot 2022-11-26 at 5 29 21 PM

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
I don't think so. The only lint errors were not related to the files for this PR.

Screen Shot 2022-11-26 at 5 50 09 PM

---

#### TODO
Create a route for the "Undo" button on the SuccessPopUp component.

### CODE REVIEW

File: src/Components/ErrorPopUp/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+import styles from './style.module.css';
+
+function ErrorPopUp(props) {
+  const [show, setShow] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('Something went wrong while processing the application status update. Please try again.');
+  const disapear = () => setShow(false);
```
Comment (renburnett) [inline src/Components/ErrorPopUp/index.js:7]: not a huge deal but should be spelled 'disappear'

Comment (renburnett) [reply src/Components/ErrorPopUp/index.js:7]: however the `setShow` hook that you created should be called directly from the `onClick` method on line 28, no need to wrap it in a function

File: src/Components/ErrorPopUp/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const { apiErrorMessage } = props;
+
+  if (apiErrorMessage) setErrorMessage(apiErrorMessage);
```
Comment (renburnett) [inline src/Components/ErrorPopUp/index.js:10]: even though it does look a bit cleaner writing your conditionals like this and its not a huge deal either way, but I think its best practice to use curly braces. 
i.e. `if (apiErrorMessage) { setErrorMessage(apiErrorMessage); }`

File: src/Components/SuccessPopUp/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+              <input
+                className={styles.undoButton}
+                type="submit"
+                value="Undo"
+              />
+              <input
+                className={styles.okayButton}
+                type="submit"
+                value="Okay"
+                onClick={() => disapear()}
```
Comment (renburnett) [inline src/Components/SuccessPopUp/index.js:31]: same thing here, we can just call `setShow(false)` instead of disappear

**Review by renburnett — CHANGES_REQUESTED** [review]
just a few small changes. and i think you opened this PR before we talked about it but try to remember to use `rems` when you can (sometimes it doesnt make sense to use like border widths etc.) anyway, looks good feel free to re-request review once changes are made ✔️

File: src/Components/SuccessPopUp/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  return (
+    <div>
+      {show && (
+        <div className={styles.container}>
+          <div className={styles.topBar}>
+            APPLICATION STATUS CHANGE
+          </div>
+          <div className={styles.mainBody}>
+            <p>
+              The application status has been successfully updated.
```
Comment (renburnett) [inline src/Components/SuccessPopUp/index.js:19]: please pass this in as a variable like you did in the ErrorPopUp. 
e.g.  `props.successMessage` 
and give this hardcoded message as the default val like you did above.

File: src/Components/ErrorPopUp/index.js (hunk 1 above)
Comment (brendobrendo) [reply src/Components/ErrorPopUp/index.js]: Oyy... Wish I could make those mistakes disappear : )

**Comment by brendobrendo** [comment]
Heads up, I took out the hooks for both successMessage and errorMessage because the way I had written was causing an infinite re-rendering loop. Instead I treated these both as normal variables. 

If you see a better and cleaner way to re-write this, I'm all ears.

Thanks Ren

File: src/Components/ErrorPopUp/index.js (hunk 1 above)
Comment (renburnett) [reply src/Components/ErrorPopUp/index.js]: haha no youre good

**Comment by renburnett** [comment]
> Heads up, I took out the hooks for both successMessage and errorMessage because the way I had written was causing an infinite re-rendering loop. Instead I treated these both as normal variables.
> 
> If you see a better and cleaner way to re-write this, I'm all ears.
> 
> Thanks Ren

nope, this is great. best practice is to keep these smaller components as dumb as possilbe

File: src/Components/ErrorPopUp/style.module.css [hunk 5]
```diff
@@ -0,0 +1,53 @@
+.container{
+  width: 500px;
```
Comment (renburnett) [inline src/Components/ErrorPopUp/style.module.css:2]: @brendobrendo change to `rem` or `percentage` equivalent of 500px

File: src/Components/ErrorPopUp/style.module.css [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+.buttonContainer{
+  padding: 30px 20px;
+  display: flex;
+  justify-content: center;
+}
+
+.okayButton{
+  width: 120px;
```
Comment (renburnett) [inline src/Components/ErrorPopUp/style.module.css:40]: @brendobrendo change to `rem` or `percentage` equivalent of 120px

File: src/Components/ErrorPopUp/style.module.css (hunk 5 above)
Comment (renburnett) [inline src/Components/ErrorPopUp/style.module.css]: @brendobrendo I see you updated the fonts to be `rems` which is great, 
but the `widths, padding and margin` should be percentages (or some other flexible value). 
we want to use as few "hard pixels" as possible for mobile-first design.

sorry I know I kinda keep harping on this, but it saves a lot of styling headaches/tweaks in the future

**Review by renburnett — APPROVED** [review]
styling updates would be nice if you can get to it, but not a huge deal.
looks great.

File: src/Components/ErrorPopUp/style.module.css (hunk 5 above)
Comment (renburnett) [reply src/Components/ErrorPopUp/style.module.css]: @brendobrendo I see you updated. looks great thanks!

**Pull request merged**